e2e-tests/cypress/tests/integration/channels/move_thread/move_thread_from_dm_spec.js

```diff
@@ -0,0 +1,189 @@
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channels @enterprise @messaging
+
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
+describe('Move Thread', () => {
+    let user1;
+    let user2;
+    let testTeam;
+    let dmChannel;
+    let testPost;
+    let replyPost;
+
+    const message = 'Move this message';
+    const replyMessage = 'Move this reply';
+
+    beforeEach(() => {
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ThreadAutoFollow: true,
+                CollapsedThreads: 'default_on',
+            },
+        });
+
+        // # Login as new user, create new team and visit its URL
+        cy.apiInitSetup({loginAfter: true, promoteNewUserAsAdmin: true}).then(({
+            user,
+            team,
+        }) => {
+            user1 = user;
+            testTeam = team;
+
+            // # enable CRT for the user
+            cy.apiSaveCRTPreference(user.id, 'on');
+
+            // # Create another user
+            return cy.apiCreateUser({prefix: 'second_'});
+        }).then(({user}) => {
+            user2 = user;
+
+            // # Add other user to team
+            return cy.apiAddUserToTeam(testTeam.id, user2.id);
+        }).then(() => {
+            // # Create new DM channel
+            return cy.apiCreateDirectChannel([user1.id, user2.id]);
+        }).then(({channel}) => {
+            dmChannel = channel;
+
+            // # Post a sample message
+            return cy.postMessageAs({sender: user1, message, channelId: dmChannel.id});
+        }).then((post) => {
+            testPost = post.data;
+
+            // # Post a reply
+            return cy.postMessageAs({sender: user1, message: replyMessage, channelId: dmChannel.id, rootId: testPost.id});
+        }).then((post) => {
+            replyPost = post.data;
+
+            // # Got to Test channel
+            cy.visit(`/${testTeam.name}/channels/${dmChannel.name}`);
+        });
+    });
+
+    afterEach(() => {
+        // # Go to 1. public channel
+        cy.visit(`/${testTeam.name}/channels/${dmChannel.name}`);
+    });
+
+    it('Move root post from DM', () => {
+        // # Check if ... button is visible in last post right side
+        cy.get(`#CENTER_button_${testPost.id}`).should('not.be.visible');
+
+        // # Click on ... button of last post
+        cy.clickPostDotMenu(testPost.id);
+
+        // * Assert availability of the Move Thread menu-item
+        cy.findByText('Move Thread').type('{shift}W');
+
+        // # move thread
+        movePostFromDM();
+
+        // * Assert switch to DM channel
+        cy.get('#channelHeaderTitle', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible').should('contain', dmChannel.display_name);
+
+        // * Assert post has been moved
+        verifyMovedThread({post: testPost});
+    });
+
+    it('Move thread reply from DM', () => {
```

Comment:
Can we add a case to test that the option to move will not be possible if `MoveThreadFromDirectMessageChannelEnable` is disabled? Same for GM and Private channels

Reply:
Added for all 3 cases